Swiftz/OptionalExt.swift

```diff
@@ -37,3 +37,25 @@ public func getOrElse<T>(m : Optional<T>) -> T -> T {
 		}
 	}
 }
+
+/// Given one or more Optional values, returns the first Optional value that is not nil
+/// when evaulated from left to right
+public func coalesce<T>(all: @autoclosure () -> T? ...) -> T? {
+	for f: () -> T? in all {
+		if let x = f() { return x }
+	}
+	return nil
+}
+
+/// Given one or more Optional values, returns the first Optional value that is not nil
+/// and satisfies the approve function when evaulated from left to right
+public func coalesce<T>(approve: T -> Bool) -> (@autoclosure () -> T? ...) -> T? {
+	return { all in
+		for f: () -> T? in all {
+			if let x = f() {
+				if approve(x) { return x }
+			}
+		}
+		return nil
+	}
+}
```

Comment:
Oh, silly me. The `and here` comments are all about the above comment about separating identifier, colon, and type with spaces.  The only newline I want is at the end of this file.